### DISCUSSION TITLE
Explicit many to many connect or create model conundrum

### DISCUSSION THREAD

**Discussion by ivandotv** [prisma/prisma#11089] [Q&A] [answered]
In the docs on [many to many relations](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations) there is this schema with explicit `m-n`
```prisma
model CategoriesOnPosts {
  post       Post     @relation(fields: [postId], references: [id])
  postId     Int // relation scalar field (used in the `@relation` attribute above)
  category   Category @relation(fields: [categoryId], references: [id])
  categoryId Int // relation scalar field (used in the `@relation` attribute above)
  assignedAt DateTime @default(now())
  assignedBy String

  @@id([postId, categoryId])
}
```
And then the post is created via:
```ts
const createCategory = await prisma.post.create({
  data: {
    title: 'How to be Bob',
    categories: {
      create: [
        {
          assignedBy: 'Bob',
          assignedAt: new Date(),
          category: {
            create: {
              name: 'New category',
            },
          },
        },
      ],
    },
  },
})
```
The post that is created is **NOT** actually connected to the category?

So is there a way to create a `post` and a `category` and connect them in one go while also checking if the category already exists?

**Comment by janpio** [comment, marked as answer]
> The post that is created is NOT actually connected to the category?

Is this an observation from when you execute the code? The post should be connected to a list of categories with some meta data (Assigned by, at) that are each connected to one category - and in this case the list has 1 entry.

**Reply by ivandotv** [reply]
I apologize, my mistake, I was looking at the wrong sqlite  file while developing.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
